jquery.flot.touchNavigate.js

```diff
@@ -230,4 +130,156 @@
+                plot.recenter({ axes: navigationState.touchedAxis });
+            }
+        }
+        navigationState.prevTouchedAxis = navigationState.currentTouchedAxis;
+        gestureState.prevTapTime = currentTime;
+    }
+
+    function zoomPlot(plot, e, gestureState, navigationState) {
+        var offset = plot.offset(),
+            center = {
+                left: 0,
+                top: 0
+            },
+            zoomAmount = pinchDistance(e) / gestureState.prevDistance,
+            dist = pinchDistance(e);
+
+        center.left = getPoint(e, 'pinch').x - offset.left;
+        center.top = getPoint(e, 'pinch').y - offset.top;
+
+        // send the computed touched axis to the zoom function so that it only zooms on that one
+        plot.zoom({
+            center: center,
+            amount: zoomAmount,
+            axes: navigationState.touchedAxis
+        });
+        gestureState.prevDistance = dist;
+    }
+
+    function wasPinchEvent(e, gestureState) {
+        return (gestureState.twoTouches && e.detail.touches.length === 1);
+    }
+
+    function getAxis(plot, e, gesture, navigationState) {
+        if (e.detail.type === 'touchstart') {
+            var point = getPoint(e, gesture);
+            return getTouchedAxis(plot, point.x, point.y);
+        } else return navigationState.touchedAxis;
+    }
+
+    function noAxisTouched(navigationState) {
+        return (!navigationState.touchedAxis || navigationState.touchedAxis.length === 0);
+    }
+
+    function setPrevDistance(e, gestureState) {
+        gestureState.prevDistance = pinchDistance(e);
+    }
+
+    function updateData(e, gesture, gestureState, navigationState) {
+        var axisDir,
+            point = getPoint(e, gesture);
+
+        switch (navigationState.navigationConstraint) {
+            case 'unconstrained':
+                navigationState.touchedAxis = null;
+                gestureState.prevTap = {
+                    x: gestureState.prevPan.x,
+                    y: gestureState.prevPan.y
+                };
+                gestureState.prevPan = {
+                    x: point.x,
+                    y: point.y
+                };
+                break;
+            case 'axisConstrained':
+                axisDir = navigationState.touchedAxis[0].direction;
+                navigationState.currentTouchedAxis = axisDir;
+                gestureState.prevTap[axisDir] = gestureState.prevPan[axisDir];
+                gestureState.prevPan[axisDir] = point[axisDir];
+                break;
+            default:
+                break;
+        }
+    }
+
+    function distance(x1, y1, x2, y2) {
+        return Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
+    }
+
+    function pinchDistance(e) {
+        var t1 = e.detail.touches[0],
+            t2 = e.detail.touches[1];
+        return distance(t1.pageX, t1.pageY, t2.pageX, t2.pageY);
```

Comment:
The pan and pinch events are 'high level' events which should make this plug-in's life easier by avoiding dealing with low level details. If possible the pan or pinch events should contain extra things like... the pinch distance.

plot.getPlaceholder()[0].childNodes[2].dispatchEvent(new CustomEvent('pinchstart', { detail: e, pinchDistance: 123 }));